Mozu.Api.Test/MsTestCases/CustomHeaderTest.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Mozu.Api.Test.Helpers;
+using System;
+using System.Collections.Generic;
+
+namespace Mozu.Api.Test.MsTestCases
+{
+    [TestClass]
+    public class CustomHeaderTest: BaseDataFactory
+    {
+        int tenantId = 0;
+
+        [TestInitialize]
+        public void TestMethodInit()
+        {
+            tenantId = Convert.ToInt32(Mozu.Api.Test.Helpers.Environment.GetConfigValueByEnvironment("TenantId"));
+        }
+
+        [TestMethod]
+        [Description("Custom Headers test")]
+        public void Http_Request_Contains_Custom_Header()
+        {
+            var customHeaders = new Dictionary<string, string>()
+            {
+                {"x-vol-cluster","unified" }
+            };
+            var baseUrlAddress = Helpers.Environment.GetConfigValueByEnvironment(Environment, "BaseAuthAppUrl");
+            SetSdKparameters();
+            var ApiMsgHandler = ServiceClientMessageFactory.GetTestClientMessage(tenantId, customHeaders);
```

Comment:
This is testing constructor instantiation. Add another test to modify existing api context via clone.

Reply:
Added couple test cases to test apiContext cloning.